### ISSUE TITLE
Error: Stream error in the HTTP/2 framing layer

### ISSUE THREAD

**Issue by baxing** [getinsomnia/insomnia#4822]
Hello 

I test GET request via Insomnia to Google Sheet Apps Script but there are some problem.
https://script.google.com/macros/s/AKfycbz2vqLC6mDxmnuR2NhefNg473JTmqcvH_eZc537Fi23z1h77owyoj5LaMoMpXxOcvupxw/exe
this link is URL for request with GET method

sometime successful
![01](https://user-images.githubusercontent.com/54997138/171011196-25c3e916-cfbe-446b-b227-cc88f146b21a.png)

and sometimes error
![02](https://user-images.githubusercontent.com/54997138/171011250-f52877b8-e90d-4d09-9b7e-dc1f79506297.png)

And with parameter it not successful request
![03](https://user-images.githubusercontent.com/54997138/171011311-ccc6d2ad-1c35-4858-a87b-d4e305467106.png)

But all requests via Postman, them successfully
 
![04](https://user-images.githubusercontent.com/54997138/171011361-6ad6474b-de89-47e2-86b8-11d9a02757bd.png)

I informed them to know the problem. Please take it into consideration in fixing the system or should I need to do some additional settings to make it work well please advise.

 
Thank you

**Comment by filfreire** [comment]
Hi @baxing, thank you for reporting this!

I suspect your problem might not be specific to Insomnia. 

You might be facing issues with [Google App Script quota limits](https://developers.google.com/apps-script/guides/services/quotas), which may help explain why sometimes the requests sometimes fail.

You also have another problem: you are wrongfully using `Multipart/Form` request body instead of using query parameters when sending the [App Scripts doGet request](https://developers.google.com/apps-script/guides/triggers/#doget_and_dopost).

This could explain why one request you are trying works on Postman (where you use query `Params`), and not on Insomnia:

![Screenshot_2022-06-02_at_12_20_38](https://user-images.githubusercontent.com/11976836/171619113-2679b8e9-d70d-4367-8cd7-dbd93e593137.png)

I'll be closing this issue for now as it's not entirely applicable to Insomnia, but let us know if after you fix this last part, the requests work afterwards. If they don't feel free to re-open the issue.

I would also like to encourage you to report any other problems you may be facing with Insomnia 👍

**Comment by baxing** [comment]
Hi filfreire,

I my cases, there are 2 problems for request :

1. GET method without parameter by url "https://script.google.com/macros/s/AKfycbz2vqLC6mDxmnuR2NhefNg473JTmqcvH_eZc537Fi23z1h77owyoj5LaMoMpXxOcvupxw/exec"

it not successful by use Insominia but it successful by use Postman. (You can try with my url in both Insominia and Postman to compare with youself)

2. GET method withparameter by url "https://script.google.com/macros/s/AKfycbz2vqLC6mDxmnuR2NhefNg473JTmqcvH_eZc537Fi23z1h77owyoj5LaMoMpXxOcvupxw/exec"

it not successful by use Insominia but it successful by use Postman. (You can try with same my url by use parameter action=read, id=2 in both Insominia and Postman to compare with youself)

Note : I try bu configure in "Qurey" instead "Multipart", It give same result

**Comment by dwedigital** [comment]
I am having the same error in Insomnia GET'ing a google function API endpoint but no issues in client side where I am running the fetch

**Comment by HarryZ10** [comment]
@dwedigital Me as well

**Comment by FlexViking** [comment]
I am having the same issue with a non Google service I am calling using Insomnia 2023.1.0 .  

- POST request works fine with no query params.  
- Adding an oData $expand query param causes the "Stream error in the HTTP/2 framing layer".  if I look at the timeline, it appears that and endless stream of chunks is being returned to me before I get "HTTP/2 stream 0 was not closed cleanly" The response is not that big ... a single chunk .. not sure why it's streaming so much ...
- Switching from HTTP/2 to HTTP/1.1 doesn't not fix the issue
- The service provider uses Postman and it all works fine.
-  If my JSON payload in the request I know is not going to match, I get a 204 response.  So, it seems like it's a combination of the oData + JSON that is causing the issue

I use oData GET requests and those don't seem to have an issue.
I am going to play with writing some Integration Tests in Java to exercise the request or some cURL commands.
Service provider thinks it has to do with URL escaping?  But I am not so sure since the GET command works fine.

**Comment by Dimaz55** [comment]
In my case the same error was because of JSON payload in GET request.

**Comment by jmevel** [comment]
Just leaving my 2 cents in case one day someone has the same issue.
I wanted to send a SOAP request to one of our services so I debugged our code and copied the XML request and pasted it into Insomnia then I stumbled on the same error message as you guys.

I checked the logs of our SOAP Service and quickly saw the problem.
For some reason the XML I copied from my debugging session had the following declaration `<?xml version="1.0" encoding="utf-16" standalone="no"?>`.

I just needed to change it to UTF-8: `<?xml version="1.0" encoding="utf-8" standalone="no"?>`

*Note that I also tried in POSTMAN and I got the error message `Error: read ECONNRESET`*

**Comment by niranjan-differenzsystem** [comment]
It is still not working. It is just annoying.

**Comment by lucastsantos** [comment]
Adding my 2 cents aswell. 
In my case, I've imported a cURL request and it had a 'Content-Length' header specified. 
This error appeared as the length specified was incorrect after changing the POST body.

**Comment by kasperhj** [comment]
In my case, deleting all cookies from "manage cookies" helped.

**Comment by humphrey** [comment]
I get this issue from time to time too.  Almost definitely a bug in Insomnia v9.3.3 because I fixed it by:

1. Changing my preferred HTTP version to HTTP 1.1 in settings
2. Retrying request (which worked)
3. Reverting the setting back to Default or HTTP 2.0 in settings
4. Retrying request (which worked)

Interestingly, before I found this fix I noticed that if I modified the url to return a 404, it worked - as it only seemed to be failing on specific 200 responses.  I also have access to the nginx server powering this particular URL - and the logs showed no issues with the request.  Here is a redacted copy of it if this helps track down the issue.

`0.0.0.0:0 - - [04/Sep/2024:10:23:01 +1000] "GET /url-that-returns-200-json-payload/ HTTP/2.0" 200 304 "-" "insomnia/9.3.3" "-" "TLSv1.3/TLS_AES_256_GCM_SHA384"`

*(Note: the response code is 200, not 302 - which just happens to be the payload size for this request)*

Update: In the time taken to write this comment, it's stopped working again and my settings hack above is no longer working.  I'm just going to have to  keep my settings on HTTP 1.1 (or install postman) to get this task I'm working on done :-(
